pkg/overlay/cache.go

```diff
@@ -82,6 +82,31 @@ func (o *Cache) Get(ctx context.Context, key string) (*overlay.Node, error) {
 	return na, nil
 }
 
+// GetAll looks up the provided nodeIDs from the overlay cache
+func (o *Cache) GetAll(ctx context.Context, keys []string) ([]*overlay.Node, error) {
+	var ks storage.Keys
+	for _, v := range keys {
+		ks = append(ks, storage.Key(v))
+	}
+	vs, err := o.DB.GetAll(ks)
+	if err != nil {
+		return nil, err
+	}
+	var ns []*overlay.Node
+	for _, v := range vs {
+		na := &overlay.Node{}
+		if v == nil {
+			ns = append(ns, nil)
+		} else if err := proto.Unmarshal(v, na); err == nil {
+			ns = append(ns, na)
+		} else {
+			//or we could append a nil value rather than returning an error. @team What do you think?
+			return nil, OverlayError.New("could not unmarshal non nil node: %s", err)
```

Comment:
re: your comment, i think this is the right behavior, but i think this err value isn't the err value from the earlier if, is it? i'm confused about the scoping.

either way, might be more clear to do something like:

```
for _, v := range vs {
  if v == nil {
    ns = append(ns, nil)
    continue
  }
  na := &overlay.Node{}
  err := proto.Unmarshal(v, na)
  if err != nil {
    return nil, OverlayError.New("could not unmarshal non-nil node: %v", err)
  }
  ns = append(ns, na)
}
```